app/controllers/internal/classified_listings_controller.rb

```diff
@@ -26,4 +27,12 @@ def destroy
   def listing_params
     params.require(:classified_listing).permit(:published, :body_markdown, :title, :category, :tag_list)
   end
+
+  def reindex_and_bust_cache
+    @classified_listing.index!
```

Comment:
```suggestion
    @classified_listing.index! if @classified_listing.published
```

Should only index if published.

Reply:
can users unpublish their listings? if they can, wouldn’t we want to re-index (or remove) it?

Reply:
Yup. We already do that in the controller.

In general this whole implementation needs to be refactored into some objects so we don't have to repeat the logic all over the place.